designs/v1-roadmap.md

```diff
@@ -0,0 +1,210 @@
+
+#### Tasks
+
+- [ ] Design and implement candidate tainting logic for NodeClaims that are past expiry, drifted, empty, etc.
+- [ ] Re-design the `karpenter.sh/disruption` taint to not differ only by value (`karpenter.sh/disruption=candidate` and `karpenter.sh/disruption=disrupting` ) but to be completely different taints so that the taints can have separate controllers managing them
+- [ ] Add the `karpenter.sh/unregistered` taint to nodes on startup to prevent pods from scheduling to the nodes while Karpenter hasn’t propagated the labels down to the nodes yet
+
+### Wait for Instance Termination on NodeClaim/Node Deletion
+
+**Issue Ref(s):** https://github.com/kubernetes-sigs/karpenter/issues/655, https://github.com/kubernetes-sigs/karpenter/issues/947
+
+**Category:** Stability
+
+Karpenter currently leaks DaemonSet pods and leases when it is terminating instances. This occurs because Karpenter currently initiates a Delete() operation once on the CloudProvider but does not continually check that the instance is fully terminated. For AWS and Azure, terminating the instance simply starts a shutdown procedure for the instance, meaning that kubelet can continue to reach out to the node until the instance is fully shutdown and terminated.
+
+Because Karpenter is not waiting, the node fails to properly deregister, leaking [daemonsets](https://github.com/kubernetes-sigs/karpenter/issues/655) and [node leases](https://github.com/aws/karpenter-provider-aws/issues/4363) onto the cluster. By waiting for instance termination before fully deleting the node, we are allowing the node to go through its graceful shutdown process.
+
+#### Tasks
+
+- [ ] Implement a retry mechanism to ensure that the instance is fully terminated before removing the Node and the NodeClaim from the cluster
+- [ ] Validate and remove the current lease garbage collection controller from `kubernetes-sigs/karpenter`, removing the permission on node leases in the `kube-node-lease` namespace
+
+### Drift Hash Breaking Change Handling
+
+**Issue Ref(s):** https://github.com/kubernetes-sigs/karpenter/issues/957
+
+**Category:** Stability
+
+Karpenter currently relies on a hash to determine whether certain fields in Karpenter’s NodeClaims have drifted from their owning NodePool and owning EC2NodeClass. Today, this is determined by hashing a set of fields on the NodePool or EC2NodeClass and then validating that this hash still matches the NodeClaim’s hash.
+
+This hashing mechanism works well for additive changes to the API, but does not work well when adding fields to the hashing function that already have a set value on a customer’s cluster. In particular, we have a need to make breaking changes to this hash scheme from these two issues: https://github.com/kubernetes-sigs/karpenter/issues/909 and https://github.com/aws/karpenter-provider-aws/issues/5447.
```

Comment:
I'm not sure I follow your question. Are you asking if a user overriding the default value will work? It will work as it normally should -- it should trigger drift and nodes should roll in response to the update